qemu/tests/uefi_boot_from_device.py

```diff
@@ -0,0 +1,92 @@
+import os
+import re
+import logging
+
+from avocado.utils import process
+
+from virttest import error_context
+from virttest import utils_misc
+from virttest import env_process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    QEMU boot from device:
+    1) Start guest from device
+       ide-hd/virtio-blk/scsi-hd/usb-storage
+       ide-cd/scsi-cd
+    2) Check the boot result
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def create_cdroms(cdrom_test):
+        """
+        Create 'test' cdrom with one file on it
+        """
+        logging.info("creating test cdrom")
+        process.run("dd if=/dev/urandom of=test bs=10M count=1")
+        process.run("mkisofs -o %s test" % cdrom_test)
+        process.run("rm -f test")
+
+    def cleanup_cdroms(cdrom_test):
+        """
+        Removes created cdrom
+        """
+        logging.info("cleaning up temp cdrom images")
+        os.remove(cdrom_test)
+
+    def boot_check(info):
+        """
+        boot info check
+        """
+        return re.search(info, vm.logsessions['seabios'].get_output(), re.S)
+
+    def count_of_move_step(boot_dev):
+        """
+        get the number of steps to move
+        """
+        logs = vm.logsessions['seabios'].get_output()
+        for i in re.findall(boot_dev, logs, re.S):
+            return len(re.findall(r"Boot\d+:\sUEFI", logs.split(i)[0], re.S))
```

Comment:
traverse a list,  return len(xxx) when handling the first item of the return of findall, so, why use 'for' loop here?

Reply:
you mean use re.findall(boot_dev, logs, re.S)[0]. The return value is a list, so I think they are all OK. I will update it.